src/jit/flowgraph.cpp

```diff
@@ -3189,9 +3189,15 @@ void Compiler::fgComputePreds()
         if (ehDsc->HasFilter())
         {
             ehDsc->ebdFilter->bbFlags |= BBF_JMP_TARGET | BBF_HAS_LABEL;
+
+            // The first block of a filter has an artifical extra refcount.
+            ehDsc->ebdFilter->bbRefs++;
```

Comment:
To elaborate a bit, these blocks start out with an extra ref count. `fgComputePreds` only increments the ref count of a BB for formal incoming edges, which these blocks do not have. The relevant code in `fgExtendEHRegionBefore` is [here](https://github.com/dotnet/coreclr/blob/master/src/jit/flowgraph.cpp#L16588-L16590) and [here](https://github.com/dotnet/coreclr/blob/master/src/jit/flowgraph.cpp#L16627-L16629).